### PR TITLE
V5.0.0 rc2

### PR DESCRIPTION
**Pull request by kevinkowa**
BREAKING CHANGE: api def tag 'sdk-major-release-2020' gen tag 3.19.0

### CODE REVIEW

File: src/IBM.Watson.Common/IBM.Watson.Common.csproj [hunk 1]
```diff
@@ -10,7 +10,7 @@
         <PackageTags>watson;cognitive;speech;vision;machine-learning;ml;ai;artificial-intelligence;.NET;.NET-Standard</PackageTags>
         <PackageIconUrl>https://watson-developer-cloud.github.io/dotnet-standard-sdk/img/Watson_Avatar_Pos_RGB.png</PackageIconUrl>
         <PackageProjectUrl>https://github.com/watson-developer-cloud/dotnet-standard-sdk</PackageProjectUrl>
-        <Version>4.6.0</Version>
+        <Version>5.0.0</Version>
```
Comment (mediumTaj) [inline src/IBM.Watson.Common/IBM.Watson.Common.csproj:13]: 👍

File: src/IBM.Watson.VisualRecognition.v4/VisualRecognitionService.cs [hunk 2]
```diff
@@ -189,16 +216,17 @@ public DetailedResponse<AnalyzeResponse> Analyze(List<string> collectionIds, Lis
         /// Encode the name and description in UTF-8 if they contain non-ASCII characters. The service assumes UTF-8
         /// encoding if it encounters non-ASCII characters.
         /// </summary>
-        /// <param name="collectionInfo">The new collection.</param>
+        /// <param name="name">The name of the collection. The name can contain alphanumeric, underscore, hyphen, and
+        /// dot characters. It cannot begin with the reserved prefix `sys-`. (optional)</param>
+        /// <param name="description">The description of the collection. (optional)</param>
+        /// <param name="trainingStatus">Training status information for the collection. (optional)</param>
```
Comment (mediumTaj) [inline src/IBM.Watson.VisualRecognition.v4/VisualRecognitionService.cs:222]: can we remove training status from here

File: src/IBM.Watson.VisualRecognition.v4/VisualRecognitionService.cs [hunk 3]
```diff
@@ -348,10 +381,17 @@ public DetailedResponse<Collection> GetCollection(string collectionId)
         /// encoding if it encounters non-ASCII characters.
         /// </summary>
         /// <param name="collectionId">The identifier of the collection.</param>
-        /// <param name="collectionInfo">The updated collection. (optional)</param>
+        /// <param name="name">The name of the collection. The name can contain alphanumeric, underscore, hyphen, and
+        /// dot characters. It cannot begin with the reserved prefix `sys-`. (optional)</param>
+        /// <param name="description">The description of the collection. (optional)</param>
+        /// <param name="trainingStatus">Training status information for the collection. (optional)</param>
```
Comment (mediumTaj) [inline src/IBM.Watson.VisualRecognition.v4/VisualRecognitionService.cs:387]: can we remove training status from here

**Review by mediumTaj — CHANGES_REQUESTED** [review]
Looking good overall, some minor changes

File: IBM.Watson.sln [hunk 4]
```diff
@@ -27,428 +27,428 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "AssistantV2", "AssistantV2"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "CompareComplyV1", "CompareComplyV1", "{B368DCF7-FB57-4302-8467-4EB142394A29}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "IBM.Watson.Assistant.v1", "src\IBM.Watson.Assistant.v1\IBM.Watson.Assistant.v1.csproj", "{A82EA32D-C38A-4C53-AA52-E97030544CEA}"
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Common", "Common", "{882BFF31-6729-459E-BCEB-556AD4CC31FA}"
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply IBM.Watson.sln:30]: it looks like this is how visual studio formats the folder structure.

File: src/IBM.Watson.VisualRecognition.v4/VisualRecognitionService.cs (hunk 3 above)
Comment (kevinkowa) [reply src/IBM.Watson.VisualRecognition.v4/VisualRecognitionService.cs]: fixed in [8b40c5f](https://github.com/watson-developer-cloud/dotnet-standard-sdk/pull/445/commits/8b40c5fad365de2a712a6bbc0b56f8aae59e6626)

File: src/IBM.Watson.Discovery.v1/Test/Integration/DiscoveryIntegrationTests.cs [hunk 5]
```diff
@@ -77,6 +77,41 @@ public void Setup()
...
+                            service.WithHeader("X-Watson-Test", "1");
+                            service.DeleteCollection(
+                                environmentId: environmentId,
+                                collectionId: collection.CollectionId
+                                );
+                            Console.WriteLine("deleted " + collection.CollectionId);
+                        }
+                        catch (Exception e)
+                        {
+                            Console.WriteLine("ERROR: Could not delete collection with id: "+collection.CollectionId);
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply src/IBM.Watson.Discovery.v1/Test/Integration/DiscoveryIntegrationTests.cs]: fixed in [960f41a](https://github.com/watson-developer-cloud/dotnet-standard-sdk/pull/445/commits/960f41abf20ab19a8dca329e30c4bd4fe512eb5d)

File: src/IBM.Watson.Discovery.v1/Test/Integration/DiscoveryIntegrationTests.cs [hunk 6]
```diff
@@ -77,6 +77,41 @@ public void Setup()
...
+
+            var collectionsList = service.ListCollections(
+                environmentId: environmentId
+                );
+
+            foreach (Collection collection in collectionsList.Result.Collections)
+            {
+                if (!string.IsNullOrEmpty(collection.Name) && !string.IsNullOrEmpty(createdCollectionName))
+                {
+                    if (collection.Name.StartsWith(createdCollectionName+"-"))
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply src/IBM.Watson.Discovery.v1/Test/Integration/DiscoveryIntegrationTests.cs]: fixed in [960f41a](https://github.com/watson-developer-cloud/dotnet-standard-sdk/pull/445/commits/960f41abf20ab19a8dca329e30c4bd4fe512eb5d)

File: src/IBM.Watson.Assistant.v2/IBM.Watson.Assistant.v2.csproj [hunk 7]
```diff
@@ -24,11 +24,19 @@
...
     </PropertyGroup>
 
+    <ItemGroup>
+      <Compile Remove="Examples\**" />
+      <Compile Remove="Test\**" />
+      <EmbeddedResource Remove="Examples\**" />
+      <EmbeddedResource Remove="Test\**" />
+      <None Remove="Examples\**" />
+      <None Remove="Test\**" />
+    </ItemGroup>
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply src/IBM.Watson.Assistant.v2/IBM.Watson.Assistant.v2.csproj:34]: yes, not sure why visual studio generated the other itemgroup

File: src/IBM.Watson.Assistant.v1/Test/Integration/IBM.Watson.Assistant.v1.IntegrationTests.csproj [hunk 8]
```diff
@@ -1,15 +1,15 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <VersionPrefix>4.6.0</VersionPrefix>
+    <VersionPrefix>5.0.0-rc2</VersionPrefix>
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply src/IBM.Watson.Assistant.v1/Test/Integration/IBM.Watson.Assistant.v1.IntegrationTests.csproj]: fixed in [b273c68](https://github.com/watson-developer-cloud/dotnet-standard-sdk/pull/445/commits/b273c68a3c4aca602dff83c9afbbd2c8533ec625)

File: src/IBM.Watson.Assistant.v1/IBM.Watson.Assistant.v1.csproj [hunk 9]
```diff
@@ -3,15 +3,15 @@
...
+        <VersionPrefix>5.0.0-rc2</VersionPrefix>
         <Authors>IBM Watson</Authors>
         <TargetFramework>netstandard2.0</TargetFramework>
         <AssemblyName>IBM.Watson.Assistant.v1</AssemblyName>
         <PackageId>IBM.Watson.Assistant.v1</PackageId>
         <PackageTags>watson;cognitive;speech;vision;machine-learning;ml;ai;artificial-intelligence;.NET;.NET-Standard</PackageTags>
         <PackageIconUrl>https://watson-developer-cloud.github.io/dotnet-standard-sdk/img/Watson_Avatar_Pos_RGB.png</PackageIconUrl>
         <PackageProjectUrl>https://github.com/watson-developer-cloud/dotnet-standard-sdk</PackageProjectUrl>
-        <Version>4.6.0</Version>
+        <Version>5.0.0-rc2</Version>
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply src/IBM.Watson.Assistant.v1/IBM.Watson.Assistant.v1.csproj]: fixed in [b273c68](https://github.com/watson-developer-cloud/dotnet-standard-sdk/pull/445/commits/b273c68a3c4aca602dff83c9afbbd2c8533ec625)

File: src/IBM.Watson.Assistant.v1/IBM.Watson.Assistant.v1.csproj [hunk 10]
```diff
@@ -3,15 +3,15 @@
     <PropertyGroup>
         <Description>IBM.Watson.Assistant.v1 wraps the Watson Assistant service (http://www.ibm.com/watson/developercloud/assistant.html)</Description>
         <AssemblyTitle>IBM.Watson.Assistant.v1</AssemblyTitle>
-        <VersionPrefix>4.6.0</VersionPrefix>
+        <VersionPrefix>5.0.0-rc2</VersionPrefix>
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply src/IBM.Watson.Assistant.v1/IBM.Watson.Assistant.v1.csproj]: fixed in [b273c68](https://github.com/watson-developer-cloud/dotnet-standard-sdk/pull/445/commits/b273c68a3c4aca602dff83c9afbbd2c8533ec625)

File: src/IBM.Watson.Assistant.v1/IBM.Watson.Assistant.v1.csproj [hunk 11]
```diff
@@ -24,11 +24,104 @@
...
+      <None Remove="Test\Integration\obj\project.nuget.cache" />
+      <None Remove="Test\Integration\obj\project.assets.json" />
+      <None Remove="Test\Integration\obj\IBM.Watson.Assistant.v1.IntegrationTests.csproj.nuget.dgspec.json" />
+      <None Remove="Test\Unit\obj\Debug\netcoreapp2.0\IBM.Watson.Assistant.v1.UnitTests.csproj.FileListAbsolute.txt" />
+      <None Remove="Test\Integration\obj\Debug\netcoreapp2.0\IBM.Watson.Assistant.v1.IntegrationTests.csproj.FileListAbsolute.txt" />
+      <None Remove="Test\Unit\obj\Debug\netcoreapp2.0\IBM.Watson.Assistant.v1.UnitTests.csproj.CoreCompileInputs.cache" />
+      <None Remove="Test\Unit\obj\Debug\netcoreapp2.0\IBM.Watson.Assistant.v1.UnitTests.AssemblyInfoInputs.cache" />
+      <None Remove="Test\Integration\obj\Debug\netcoreapp2.0\IBM.Watson.Assistant.v1.IntegrationTests.csproj.CoreCompileInputs.cache" />
+      <None Remove="Test\Unit\obj\Debug\netcoreapp2.0\IBM.Watson.Assistant.v1.UnitTests.csprojAssemblyReference.cache" />
+    </ItemGroup>
```
[the remark this answers is not in the document]
Comment (kevinkowa) [reply src/IBM.Watson.Assistant.v1/IBM.Watson.Assistant.v1.csproj]: it's fine without them, fixed in [878be7d](https://github.com/watson-developer-cloud/dotnet-standard-sdk/pull/445/commits/878be7d6ac55e7228bd0fee250475965da58d06d)

**Review by mediumTaj — APPROVED** [review]
👍 looks good!

**Review by mediumTaj — APPROVED** [review]

File: src/IBM.Watson.VisualRecognition.v3/VisualRecognitionService.cs [hunk 12]
```diff
@@ -28,23 +32,27 @@
 
 namespace IBM.Watson.VisualRecognition.v3
 {
+    [System.Obsolete("On 1 December 2021, Visual Recognition will no longer be available. " +
+        "For more information, see Visual Recognition Deprecation " +
+        "(https://github.com/watson-developer-cloud/java-sdk/tree/master#visual-recognition-deprecation).")]
```
Comment (mamoonraja) [inline src/IBM.Watson.VisualRecognition.v3/VisualRecognitionService.cs:37]: link is pointing to `java-sdk`

Comment (kevinkowa) [reply src/IBM.Watson.VisualRecognition.v3/VisualRecognitionService.cs:37]: good catch!

Comment (kevinkowa) [reply src/IBM.Watson.VisualRecognition.v3/VisualRecognitionService.cs]: fixed!

Comment (mamoonraja) [reply src/IBM.Watson.VisualRecognition.v3/VisualRecognitionService.cs]: no worries, I almost did the same for unity at first 😂

**Pull request merged**